forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/test/java/com/forgerock/openbanking/aspsp/as/api/authorisation/AuthorisationApiControllerTest.java

```diff
@@ -62,96 +70,126 @@
+                null, "payments openid accounts", null, jwt, true, null, null, null, null, null);
 
         // Then no exception
     }
 
     @Test
-    public void shouldNotThrowExceptionWhenNoUserInfo() throws OBErrorException, OBErrorResponseException {
+    public void shouldNotThrowExceptionWhenNoUserInfo() throws OBErrorException, OBErrorResponseException,
+            InvalidTokenException, ParseException, IOException {
         // Given
+        String clientId = "98e119f6-196f-4296-98d4-f1a2f445bca2";
         List<String> responseTypes = List.of("code id_token");
         given(discoveryConfig.getSupportedResponseTypes()).willReturn(responseTypes);
         String jwt = toEncodedSignedTestJwt("jwt/authorisation-no-user-info.jwt");
         Tpp tpp = new Tpp();
         OIDCRegistrationResponse registrationResponse = new OIDCRegistrationResponse();
-        registrationResponse.setJwks_uri("");
+        registrationResponse.setJwks_uri("url");
         tpp.setRegistrationResponse(registrationResponse);
-        given(tppStoreService.findByClientId(null)).willReturn(Optional.of(tpp));
+        given(tppStoreService.findByClientId(clientId)).willReturn(Optional.of(tpp));
+        SignedJWT signedJwt = mock(SignedJWT.class);
+        given(cryptoApiClient.validateJws(anyString(), anyString(), anyString() )).willReturn(signedJwt);
 
         // When
-        authorisationApiController.getAuthorisation(responseTypes.get(0), null, null, null, "payments openid accounts", null, jwt, true, null, null, null, null, null);
+        authorisationApiController.getAuthorisation(responseTypes.get(0), clientId, null,
+                null, "payments openid accounts", null, jwt, true, null, null, null, null, null);
 
         // Then no exception
     }
 
     @Test(expected = OBErrorException.class)
-    public void shouldThrowExceptionWhenJwtScopesDoNotMatchQueryParamScope() throws OBErrorException, OBErrorResponseException {
+    public void shouldThrowExceptionWhenJwtScopesDoNotMatchQueryParamScope()
+            throws OBErrorException, OBErrorResponseException {
         // Given
         List<String> responseTypes = List.of("code id_token");
         given(discoveryConfig.getSupportedResponseTypes()).willReturn(responseTypes);
         String jwt = toEncodedSignedTestJwt("jwt/authorisation.jwt");
-        Tpp tpp = new Tpp();
         OIDCRegistrationResponse registrationResponse = new OIDCRegistrationResponse();
-        registrationResponse.setJwks_uri("");
-        tpp.setRegistrationResponse(registrationResponse);
-        given(tppStoreService.findByClientId(null)).willReturn(Optional.of(tpp));
+        registrationResponse.setJwks_uri("url");
 
         // When
-        authorisationApiController.getAuthorisation(responseTypes.get(0), null, null, null, "openid accounts", null, jwt, true, null, null, null, null, null);
+        authorisationApiController.getAuthorisation(responseTypes.get(0), "98e119f6-196f-4296-98d4-f1a2f445bca2", null,
```

Comment:
It's better to assert the exception using `catchThrowableOfType()` (like you have in some of the other tests in this PR).

Reply:
I didn't wroite this test, but yeah. It would be better.

Reply:
Fixed